### PR TITLE
[PW-6556] - Modification of usages of strpos() & trim() for PHP 8.1

### LINKED ISSUE #1472: Exception: Deprecated Functionality: trim(): Passing null to parameter #1 ($string) of type string is deprecated in /var/www/html/vendor/adyen/module-payment/Helper/PaymentMethods.php on line 376
**Deprecated Functionality: trim() error on php 8**
When we install Adyen 8.2.1 on magento 2.4.4 with PHP 8, it throws below error

Exception: Deprecated Functionality: trim(): Passing null to parameter #1 ($string) of type string is deprecated in /var/www/html/vendor/adyen/module-payment/Helper/PaymentMethods.php on line 376

**To Reproduce**
1. Install Adyen 8.2.1 module on magento 2.4.4. with PHP 8
2. Go to checkout page

**Expected behavior**
Checkout page should load without any errors

**Magento version**
2.4.4

**Plugin version**
8.2.1

**Screenshots**
[2022-04-22T14:05:19.248513+00:00] report.CRITICAL: Exception: Deprecated Functionality: trim(): Passing null to parameter #1 ($string) of type string is deprecated in /var/www/html/vendor/adyen/module-payment/Helper/PaymentMethods.php on line 376 in /var/www/html/vendor/magento/framework/App/ErrorHandler.php:61

**Desktop (please complete the following information):**
 - Ubuntu
 - Chrome
 - Version 100.0.4896.127 (Official Build) (64-bit)

**Additional context**
[2022-04-22T14:05:19.248513+00:00] report.CRITICAL: Exception: Deprecated Functionality: trim(): Passing null to parameter #1 ($string) of type string is deprecated in /var/www/html/vendor/adyen/module-payment/Helper/PaymentMethods.php on line 376 in /var/www/html/vendor/magento/framework/App/ErrorHandler.php:61

### PR DESCRIPTION
**Pull request by candemiralp**
**Description**

Passing null argument parameter to the following functions deprecated in PHP 8.0

- trim()
- strpos() 
- date_create_from_format()

Possible null arguments were checked and returned false in the required cases.

**Tested scenarios**

1. Cards payments
2. Redirect LPM
3. Paybylink

**Closes issues**
Closes #1472

### CODE REVIEW

**Review by michaelpaul — CHANGES_REQUESTED** [review]
Adding null coalesce solves the problem but hides the error/fail path. Just adding the empty string will only propagate the error to `trim` and `decrypt`, which will fail as well with such input:   
```php
$secretWord = $this->_encryptor->decrypt(trim($this->getAdyenHppConfigData('hmac_live', $storeId) ?? ''));
```

I think we should be more explicit about these flows:
```php
if (is_null($key)) {
    return '';
}
```

File: Helper/Config.php [hunk 1]
```diff
@@ -112,6 +112,10 @@ public function getNotificationsPassword($storeId = null)
             self::XML_ADYEN_ABSTRACT_PREFIX,
             $storeId
         );
+
+        if (is_null($key)) {
+            return false;
```
Comment (peterojo) [inline Helper/Config.php:117]:
```suggestion
            return null;
```

File: Helper/Config.php [hunk 2]
```diff
@@ -186,6 +190,11 @@ public function getNotificationsHmacKey($storeId = null)
                 false
             );
         }
+
+        if (is_null($key)) {
+            return false;
```
Comment (peterojo) [inline Helper/Config.php:195]:
```suggestion
            return null;
```

File: Helper/Data.php [hunk 3]
```diff
@@ -47,6 +47,7 @@
 use Magento\Store\Model\StoreManagerInterface;
 use Magento\Tax\Model\Calculation;
 use Magento\Tax\Model\Config;
+use function PHPUnit\Framework\isNull;
```
Comment (peterojo) [inline Helper/Data.php:50]: why is this necessary?

File: Observer/AdyenSalesOrderChargedCurrencyObserver.php [hunk 4]
```diff
@@ -47,6 +47,11 @@ public function execute(Observer $observer)
         /** @var \Magento\Sales\Model\Order $order */
         $order = $observer->getEvent()->getOrder();
         $paymentMethod = $order->getPayment()->getMethod();
+
+        if (is_null($paymentMethod)) {
+            return false;
```
Comment (peterojo) [inline Observer/AdyenSalesOrderChargedCurrencyObserver.php:52]: Return type for `execute` is void
```suggestion
            return;
```

File: Gateway/Validator/PayByLinkValidator.php [hunk 5]
```diff
@@ -34,11 +34,11 @@ class PayByLinkValidator extends AbstractValidator
...
     {
         $payment = $validationSubject['payment'];
-        $expiryDate = date_create_from_format(
-            AdyenPayByLinkConfigProvider::DATE_FORMAT,
-            $payment->getAdyenPblExpiresAt()
-        );
-        if ($expiryDate) {
+
+        if (
+            !is_null($payment->getAdyenPblExpiresAt()) &&
```
[the remark this answers is not in the document]
Comment (candemiralp) [reply Gateway/Validator/PayByLinkValidator.php:39]: Thank you @michaelpaul

File: Helper/Data.php (hunk 3 above)
Comment (candemiralp) [reply Helper/Data.php:50]: Just a mistake due to the auto completion of Phpstorrm.

File: Observer/AdyenSalesOrderChargedCurrencyObserver.php (hunk 4 above)
Comment (candemiralp) [reply Observer/AdyenSalesOrderChargedCurrencyObserver.php:52]: Thank you for your comments @peterojo

File: Helper/Data.php [hunk 6]
```diff
@@ -613,10 +613,18 @@ public function getHmac($storeId = null)
     {
         switch ($this->isDemoMode($storeId)) {
             case true:
-                $secretWord = $this->_encryptor->decrypt(trim($this->getAdyenHppConfigData('hmac_test', $storeId)));
+                $hmacTest = $this->getAdyenHppConfigData('hmac_test', $storeId);
+                if (is_null($hmacTest)) {
+                    return false;
```
Comment (michaelpaul) [inline Helper/Data.php:618]: Let's turn `string` configurations into `string|null` to represent lack of setup:
```suggestion
                    return null;
```

File: Helper/Data.php [hunk 7]
```diff
@@ -643,23 +651,17 @@ public function isDemoMode($storeId = null)
...
-                trim(
-                    $this->getAdyenAbstractConfigData(
-                        'api_key_live',
-                        $storeId
-                    )
-                )
-            );
+            $encryptedApiKeyLive = $this->getAdyenAbstractConfigData('api_key_live', $storeId);
+            if (is_null($encryptedApiKeyLive)) {
+                return false;
```
Comment (michaelpaul) [inline Helper/Data.php:662]: Let's turn `string` configurations into `string|null` to represent lack of setup:
```suggestion
                return null;
```

File: Helper/Data.php [hunk 8]
```diff
@@ -672,12 +674,16 @@ public function getAPIKey($storeId = null)
...
-                $this->isDemoMode($storeId) ? 'client_key_test' : 'client_key_live',
-                $storeId
-            )
+        $clientKey = $this->getAdyenAbstractConfigData(
+            $this->isDemoMode($storeId) ? 'client_key_test' : 'client_key_live',
+            $storeId
         );
+
+        if (is_null($clientKey)) {
+            return false;
```
Comment (michaelpaul) [inline Helper/Data.php:683]: Let's turn `string` configurations into `string|null` to represent lack of setup:
```suggestion
            return null;
```

File: Helper/Data.php [hunk 9]
```diff
@@ -689,9 +695,17 @@ public function getClientKey($storeId = null)
     public function getWsUsername($storeId = null)
     {
         if ($this->isDemoMode($storeId)) {
-            $wsUsername = trim($this->getAdyenAbstractConfigData('ws_username_test', $storeId));
+            $wsUsernameTest = $this->getAdyenAbstractConfigData('ws_username_test', $storeId);
+            if (is_null($wsUsernameTest)) {
+                return false;
```
Comment (michaelpaul) [inline Helper/Data.php:700]: Let's turn `string` configurations into `string|null` to represent lack of setup:
```suggestion
                return null;
```

File: Helper/Data.php [hunk 10]
```diff
@@ -689,9 +695,17 @@ public function getClientKey($storeId = null)
...
+            }
+            $wsUsername = trim($wsUsernameTest);
         } else {
-            $wsUsername = trim($this->getAdyenAbstractConfigData('ws_username_live', $storeId));
+            $wsUsernameLive = $this->getAdyenAbstractConfigData('ws_username_live', $storeId);
+            if (is_null($wsUsernameLive)) {
+                return false;
```
Comment (michaelpaul) [inline Helper/Data.php:706]: Let's turn `string` configurations into `string|null` to represent lack of setup:
```suggestion
                return null;
```

File: Helper/Data.php [hunk 11]
```diff
@@ -704,8 +718,13 @@ public function getWsUsername($storeId = null)
      */
     public function getLiveEndpointPrefix($storeId = null)
     {
-        $prefix = trim($this->getAdyenAbstractConfigData('live_endpoint_url_prefix', $storeId));
-        return $prefix;
+        $prefix = $this->getAdyenAbstractConfigData('live_endpoint_url_prefix', $storeId);
+
+        if (is_null($prefix)) {
+            return false;
```
Comment (michaelpaul) [inline Helper/Data.php:724]: Let's turn `string` configurations into `string|null` to represent lack of setup:
```suggestion
            return null;
```

File: Helper/Data.php [hunk 12]
```diff
@@ -716,8 +735,13 @@ public function getLiveEndpointPrefix($storeId = null)
      */
     public function getCheckoutFrontendRegion($storeId = null)
     {
-        $prefix = trim($this->getAdyenAbstractConfigData('checkout_frontend_region', $storeId));
-        return $prefix;
+        $checkoutFrontendRegion = $this->getAdyenAbstractConfigData('checkout_frontend_region', $storeId);
+
+        if (is_null($checkoutFrontendRegion)) {
+            return false;
```
Comment (michaelpaul) [inline Helper/Data.php:741]: Let's turn `string` configurations into `string|null` to represent lack of setup:
```suggestion
            return null;
```

File: Helper/Data.php [hunk 13]
```diff
@@ -1388,9 +1421,19 @@ public function getOpenInvoiceLineData(
     public function getPosApiKey($storeId = null)
     {
         if ($this->isDemoMode($storeId)) {
-            $apiKey = $this->_encryptor->decrypt(trim($this->getAdyenPosCloudConfigData('api_key_test', $storeId)));
+            $encryptedApiKeyTest = $this->getAdyenPosCloudConfigData('api_key_test', $storeId);
+            if (is_null($encryptedApiKeyTest)) {
+                return false;
```
Comment (michaelpaul) [inline Helper/Data.php:1426]: Let's turn `string` configurations into `string|null` to represent lack of setup:
```suggestion
                return null;
```

File: Helper/Data.php [hunk 14]
```diff
@@ -1388,9 +1421,19 @@ public function getOpenInvoiceLineData(
...
+            }
+
+            $apiKey = $this->_encryptor->decrypt(trim($encryptedApiKeyTest));
         } else {
-            $apiKey = $this->_encryptor->decrypt(trim($this->getAdyenPosCloudConfigData('api_key_live', $storeId)));
+            $encryptedApiKeyLive = $this->getAdyenPosCloudConfigData('api_key_live', $storeId);
+            if (is_null($encryptedApiKeyLive)) {
+                return false;
```
Comment (michaelpaul) [inline Helper/Data.php:1433]: Let's turn `string` configurations into `string|null` to represent lack of setup:
```suggestion
                return null;
```

File: Plugin/PaymentVaultDeleteToken.php [hunk 15]
```diff
@@ -54,7 +54,9 @@ public function beforeDelete(
         \Magento\Vault\Api\PaymentTokenRepositoryInterface $subject,
         PaymentTokenInterface $paymentToken
     ) {
-        if (strpos($paymentToken->getPaymentMethodCode(), 'adyen_') !== 0) {
+        $paymentMethodCode = $paymentToken->getPaymentMethodCode();
+
+        if (!is_null($paymentMethodCode) && strpos($paymentMethodCode, 'adyen_') !== 0) {
```
Comment (michaelpaul) [inline Plugin/PaymentVaultDeleteToken.php:59]: When there is no payment method or it's not Adyen we can return early: 
```suggestion
        if (is_null($paymentMethodCode) || strpos($paymentMethodCode, 'adyen_') !== 0) {
```

**Review by michaelpaul — CHANGES_REQUESTED** [review]

**Review by michaelpaul — APPROVED** [review]

**Review by peterojo — APPROVED** [review]

**Pull request merged**